wicker/core/datasets.py

```diff
@@ -141,44 +245,69 @@ def _get_parquet_dir_size(self) -> int:
         # bytes in arrow table is a method of arrow table but it doesn't
         # reflect the size of the file sizes stored on s3 just the loaded data
         arrow_path = self._s3_path_factory.get_dataset_partition_path(self._partition, s3_prefix=False)
+
+        # have to cut off the end because we want entire dir size not just one partition
+        # the entire dataset is all parquet files
         bucket, key = arrow_path.replace("s3://", "").split("/", 1)
```

Comment:
@pickles-bread-and-butter it seems like the original intention was just to get the dataset partition size (i.e. "training", "evaluation", "validation", etc.) so that if you want to get the entire dataset size, you should create separate `S3Dataset` objects and calculate and sum all their sizes. This seems a little bit painful but fine.

One option if you want to compute the entire dataset size in one pass is to add an extra optional argument to this function that declares if you want the entire dataset size (and if so do the extra key trimming you suggested).

Reply:
I wrote the original function to like you said, get the size of the entire dataset by parsing through the column concatenated files. Yeah the pain point is what you mentioned, there's no way to know all the partitions without seeing that folder. Another pain point of Wickers essentially non-existant metadata.

Reply:
@pickles-bread-and-butter easy solution, could you add an `Optional` boolean parameter to the function to tell it to compute the size of the entire dataset (which should probably be `False` by default)? If computing the size of the entire dataset, perform the key splitting! Move the comment right above next to the if-condition check on this flag and the key splitting. I think that is a good solution that preserves the readability of the function while still being convenient.

Reply:
I'm just thinking a simple flag that either does the key splitting in the next line or not basically. I guess if the key splitting is skipped, that is equivalent to reading only the partition selected by the user.

I'm not too concerned about all this, I was just thinking that adding a flag to the function arguments and then checking it here might help any future readers of the code understand what is happening.